packages/admin-ui/src/number-input/number-input.state.ts

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState } from 'react'
+import type { ButtonProps } from '../button'
+import {
+  isNumber,
+  isNumberAsString,
+  getIncrementedValue,
+  getDecrementedValue,
+} from './utils'
+
+export const useNumberInput = (props: NumberInputState) => {
+  const { initialValue, step, min, max, onChange } = props
+  const [currentValue, setCurrentValue] = useState(initialValue)
+
+  const handleChange: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+    const { value } = event.target
+    const newValue = isNumberAsString(value) ? Number(value) : value
```

Comment:
I think you should add a validation here to check if the input is bigger or lower than the min and max values. I'm not sure if this is expected:

![Captura de Tela 2022-05-24 às 16 31 02](https://user-images.githubusercontent.com/20579226/170117184-cb5e93c2-0a85-4e21-9d84-6913085801a7.png)